packages/qwik/src/optimizer/src/platform.ts

```diff
@@ -265,10 +265,6 @@ const getEnv = (): SystemEnvironment => {
     return 'deno';
   }
 
-  if (typeof Bun !== 'undefined') {
-    return 'bun';
```

Comment:
is there a way to make the pr work without doing this?

Reply:
There should be a better way, the problem is likely on my side, because I have no full understanding of the code. I tried to add test the result of this function in where it is used, but this way it fails.

More specifically, here (this is where getSystem is called and then the platform is tested in if statements):

https://github.com/QwikDev/qwik/blob/a6a8525cd676c6e9c8e92ddf00466635d6c78c47/packages/qwik/src/optimizer/src/platform.ts#L38

https://github.com/QwikDev/qwik/blob/a6a8525cd676c6e9c8e92ddf00466635d6c78c47/packages/qwik/src/optimizer/src/platform.ts#L65

https://github.com/QwikDev/qwik/blob/a6a8525cd676c6e9c8e92ddf00466635d6c78c47/packages/qwik/src/optimizer/src/platform.ts#L167

https://github.com/QwikDev/qwik/blob/a6a8525cd676c6e9c8e92ddf00466635d6c78c47/packages/qwik/src/optimizer/src/platform.ts#L228

Then getSystem function is called here:

https://github.com/QwikDev/qwik/blob/a6a8525cd676c6e9c8e92ddf00466635d6c78c47/packages/qwik/src/optimizer/src/optimizer.ts#L17

And the result from `getEnv` function is used here:

https://github.com/QwikDev/qwik/blob/a6a8525cd676c6e9c8e92ddf00466635d6c78c47/packages/qwik/src/optimizer/src/plugins/plugin.ts#L309

and here

https://github.com/QwikDev/qwik/blob/a6a8525cd676c6e9c8e92ddf00466635d6c78c47/packages/qwik/src/optimizer/src/plugins/plugin.ts#L331C19-L331C37

It didn't work when I added bun to these tests and tried this locally, so I went with this solution. I definitely missed something.

Reply:
this is the only change preventing me from merging. can you just add if node or bun. rather than just removing bun

Reply:
This is the result with current solution:

![telegram-cloud-document-2-5362005861461937410](https://github.com/QwikDev/qwik/assets/7884558/dd747a81-2e70-4434-812f-1a1d48ecfff4)

I will try again anyway, because what I did isn't good in any way, but it works.

Reply:
Oh, ok. Maybe I didn't tested properly. It works. But shouldn't I also change plugin.ts same way?